packages/cli/src/commands/labs/index.ts

```diff
@@ -0,0 +1,87 @@
+import color from '@heroku-cli/color'
+import {Command, flags} from '@heroku-cli/command'
+import * as Heroku from '@heroku-cli/schema'
+import {Args, ux} from '@oclif/core'
+import {sortBy} from 'lodash'
+
+interface Features {
+  currentUser: any,
+  user: any,
+  app?: any,
+}
+
+function printJSON(features: Heroku.Account | Heroku.AccountFeature | Heroku.AppFeature) {
+  ux.log(JSON.stringify(features, null, 2))
+}
+
+function printFeatures(features: any) {
+  features = sortBy(features, 'name')
+  const longest = Math.max.apply(null, features.map((f: Record<string, string>) => f.name.length))
+  for (const f of features) {
+    let line = `${f.enabled ? '[+]' : '[ ]'} ${f.name!.padEnd(longest)}`
+    if (f.enabled) line = color.green(line)
+    line = `${line}  ${f.description}`
+    ux.log(line)
+  }
+}
+
+export default class LabsIndex extends Command {
+  static description = 'list experimental features'
+  static topic = 'labs'
+
+  static args = {
+    app: Args.string({required: false}),
+  }
+
+  static flags = {
+    app: flags.app({required: false}),
+    json: flags.boolean({description: 'display as json', required: false}),
+  }
+
+  async run() {
+    const {args, flags} = await this.parse(LabsIndex)
+    const [currentUserResponse, userResponse, appResponse] = await Promise.all([
+      this.heroku.get<Heroku.Account>('/account'),
+      this.heroku.get<Heroku.AccountFeature>('/account/features'),
+      (args.app || flags.app) ? this.heroku.get<Heroku.AppFeature>(`/apps/${args.app ? args.app : flags.app}/features`) : null,
```

Comment:
@ryandagg  I like the approach with conditionally pushing the request to the promises array, but it removes the ability to use the ternary operator necessary to default `app` to null if neither `args.app` or `flags.app` are present. Deconstructing the promise is where everything becomes more complicated than the current code since you can't optionally deconstruct an expected response shape. I'm sure we could pick apart the array, but I don't think it's worth the additional logic.